src/compas/files/base_reader.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+    def __init__(self, location):
+        self.is_url = False
+        self.location = self.check_location(location)
```

Comment:
would this not be a perfect use case for `@property`?

Reply:
Like this?

https://github.com/compas-dev/compas/blob/d27f338942d4d5109ad888c80534bd1ca8398193/src/compas/files/base_reader.py#L40